### PR TITLE
solution to well-formed-strings challenge

### PR DESCRIPTION
**Pull request by bermannoah**
- checks strings + validates that they are well formed

### CODE REVIEW

**Comment by jcasimir** [comment]
Tests?

**Comment by bermannoah** [comment]
Whoops! Just pushed 'em. @jcasimir

File: well-formed.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'pry'
+
+class StringValidator
+
+  def validate(string)
+    trimmed_string = string.gsub(/[^\[\]\(\)\{\}]/,'')
```
Comment (mikedao) [inline well-formed.rb:6]: Interesting approach here. How flexible is it if we added other symbols?

File: well-formed.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      return false
+    elsif trimmed_string.empty?
+      return true
+    else
+      return even_string_validator(string)
+    end
+  end
+  
+  def even_string_validator(string)
+    symbols_to_check = { '{' => '}',
```
Comment (mikedao) [inline well-formed.rb:17]: Is there a better way to store this to make it more reusable? Its this mutable? Should it be?

File: well_formed_test.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     result = @v.validate("({12)}")
+     assert_equal result, false
+   end
+
+
+   def test_well_formed_test_case_with_gibberish
+     result = @v.validate("({12oiuh@#((((1213)))))}lol}")
+     assert_equal result, false
+   end
+
```
Comment (mikedao) [inline well_formed_test.rb:64]: So much testing! Very good.

**Review by mikedao — APPROVED** [review]

**Pull request closed without merging**